File: scripts/fetch-butt-covers.js
const axios = require('axios');
const cheerio = require('cheerio');
const fs = require('fs').promises;
const path = require('path');

async function fetchButtCovers() {
  const baseUrl = 'https://buttmagazine.com/library/';
  const issueData = [];

  console.log('Fetching Butt Magazine library page...');

  try {
    // Fetch the main library page
    const response = await axios.get(baseUrl);
    const $ = cheerio.load(response.data);

    // Find all issue links
    $('a[href*="/library/butt-"]').each((i, elem) => {
      const href = $(elem).attr('href');
      const issueMatch = href.match(/butt-(\d+)/);

      if (issueMatch) {
        const issueNum = parseInt(issueMatch[1]);
        if (issueNum >= 3 && issueNum <= 37) {
          // Try to find image in the link
          const img = $(elem).find('img');
          let imgSrc = img.attr('src') || img.attr('data-src') || img.attr('data-lazy-src');

          issueData.push({
            issue: issueNum,
            url: href,
            imageSrc: imgSrc
          });
        }
      }
    });

    // Sort by issue number
    issueData.sort((a, b) => a.issue - b.issue);

    console.log(`\nFound ${issueData.length} issues:`);
    issueData.forEach(item => {
      console.log(`Issue #${item.issue}: ${item.imageSrc || 'No image found'}`);
    });

    // Save to JSON
    await fs.writeFile(
      path.join(__dirname, 'butt-covers.json'),
      JSON.stringify(issueData, null, 2)
    );

    console.log('\nSaved to scripts/butt-covers.json');

  } catch (error) {
    console.error('Error:', error.message);
  }
}

fetchButtCovers();

File: scripts/fetch-butt-covers.py
#!/usr/bin/env python3
import json
import asyncio
from playwright.async_api import async_playwright

async def fetch_butt_covers():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        page = await browser.new_page()

        print("Fetching Butt Magazine library page...")
        await page.goto('https://buttmagazine.com/library/', wait_until='networkidle')

        # Wait for images to load
        await page.wait_for_timeout(3000)

        # Extract issue data
        issues = await page.evaluate('''() => {
            const issueData = [];
            const links = Array.from(document.querySelectorAll('a[href*="/library/butt-"]'));

            links.forEach(link => {
                const href = link.getAttribute('href');
                const match = href.match(/butt-(\\d+)/);

                if (match) {
                    const issueNum = parseInt(match[1]);
                    if (issueNum >= 3 && issueNum <= 37) {
                        const img = link.querySelector('img');
                        const imgSrc = img ? (img.src || img.getAttribute('data-src') || img.getAttribute('srcset')) : null;

                        issueData.push({
                            issue: issueNum,
                            url: href,
                            imageSrc: imgSrc
                        });
                    }
                }
            });

            // Remove duplicates and sort
            const unique = Array.from(new Map(issueData.map(item => [item.issue, item])).values());
            return unique.sort((a, b) => a.issue - b.issue);
        }''')

        print(f"\nFound {len(issues)} issues:")
        for item in issues:
            print(f"Issue #{item['issue']}: {item['imageSrc'][:100] if item['imageSrc'] else 'No image found'}...")

        # Save to JSON
        with open('scripts/butt-covers.json', 'w') as f:
            json.dump(issues, f, indent=2)

        print("\nSaved to scripts/butt-covers.json")

        await browser.close()

if __name__ == '__main__':
    asyncio.run(fetch_butt_covers())
